evm_arithmetization/src/generation/mod.rs

```diff
@@ -348,3 +305,60 @@ fn simulate_cpu<F: Field>(state: &mut GenerationState<F>) -> anyhow::Result<()>
 
     Ok(())
 }
+
+/// Outputs the tries that have been obtained post transaction execution, as
+/// they are represented in the prover's memory.
+/// This will do nothing if the CPU execution failed outside of the final trie
+/// root checks.
+pub(crate) fn output_debug_tries<F: RichField>(state: &GenerationState<F>) -> anyhow::Result<()> {
+    if !log_enabled!(log::Level::Debug) {
+        return Ok(());
+    }
+
+    // Retrieve previous PC (before jumping to KernelPanic), to see if we reached
+    // `hash_final_tries`. We will output debugging information on the final
```

Comment:
nit: I think you changed `hash_final_tries` --> `perform_final_checks` (which is a much better name).

Reply:
Oops, yes I did, thanks!